Userland/Libraries/LibGL/GLContext.h

```diff
@@ -91,6 +91,14 @@ class GLContext final {
     NonnullRefPtr<Gfx::Bitmap> frontbuffer() const { return m_frontbuffer; };
     void present();
 
+    // Used by WebGL to preserve the clear values when implicitly clearing the front buffer.
+    FloatVector4 current_clear_color() const { return m_clear_color; }
+    GLdouble current_clear_depth() const { return m_clear_depth; }
+    GLint current_clear_stencil() const { return m_clear_stencil; }
+
+    // Used by WebGL to implement WebGL specific restrictions.
+    void set_error(GLenum error);
```

Comment:
This might be confused for a generic setter that always sets an error code; maybe clarify this in the method name? Not sure what that the best candidate would be, `set_error_if_no_error_set` sounds a bit :yakid2:

Reply:
Feels a bit hackish actually 🤔 Would this not better be placed directly in the WebGL context? The GL context has no idea about WebGL specific error conditions and it's a big state machine. I feel that's just asking for trouble.